integrationExamples/gpt/esp_example.html

```diff
@@ -0,0 +1,210 @@
+                                expires: 90,             // storage lasts for 90 days
+                                refreshInSeconds: 8 * 3600 // refresh ID every 8 hours to ensure it's fresh
+                            }
+                        },{
+                            name: "sharedId",
+                            storage: {
+                                type: "cookie",
+                                name: `_pubcid`,         // create a cookie with this name
+                                expires: 365             // expires in 1 years
+                            }
+                        },{
+                            name: "criteo",
+                        },{
+                            name: "pubCommonId",
+                            storage: {
+                                type: "cookie",
+                                name: "_pubcid",         // create a cookie with this name
+                                expires: 365             // expires in 1 years
+                            }
+                        },{
+                            name: 'hadronId',
+                            storage: {
+                                name: 'hadronId',
+                                type: 'html5'
+                            }
+                        }],
+                        auctionDelay: 10000
+                    }
+                });
+                pbjs.requestBids({
+                    bidsBackHandler: initAdserver,
+                    timeout: PREBID_TIMEOUT
+                });
+            });
+
+
+            
+
+            function initAdserver() {
+                if (pbjs.initAdserverSet) return;
+                pbjs.initAdserverSet = true;
+                googletag.cmd.push(function() {
+                    pbjs.que.push(function() {
+                        pbjs.setTargetingForGPTAsync();
+                        googletag.encryptedSignalProviders= [];
+                        // Timeout added for now to get eids from all the partners before registering the signal
+                        setTimeout(function(){
+                            var signalSources =  [  
+                                "sharedid.org", 
+                                "criteo.com",
+                                "id5-sync.com", 
+                                "pubcid.org", 
+                                "audigent.com"
+                            ];
+                            var encrypt = false;
+                            //Register signal sources
+                            pbjs.registerSignalSources(googletag, signalSources, encrypt);
+                            //Register custom signals
+                            var customSignalSources = ["pubmatic.com", "mysegments.com"];
+                            pbjs.registerSignalSources(googletag, customSignalSources, encrypt, getCustomSignal);
+                            //googletag.pubads().refresh();
+                        },3000)
```

Comment:
Just curious, what is the main purpose of the timeout here?

Reply:
@r-schweitzer This was added for the workaround purpose to solve the caching the null values issues in storage for some partners
By adding a timeout, it will register the signals after all the partners' calls are done though cannot guarantee it. The Null issue observation was shared under description. We can go ahead without timeout too but it will cache nulls value for registered signals if partner calls are delayed. Cached null signals won't be refreshed until the next expiry.

Reply:
Resolved: Timeout added as encryptedSignal configs which is set in userSync object.